fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/navbar/NavigationBarButtons.kt

```diff
@@ -0,0 +1,206 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.toolbar.navbar
+
+import androidx.navigation.NavController
+import mozilla.components.browser.menu.view.MenuButton
+import mozilla.components.browser.thumbnails.BrowserThumbnails
+import mozilla.components.support.base.feature.ViewBoundFeatureWrapper
+import org.mozilla.fenix.GleanMetrics.StartOnHome
+import org.mozilla.fenix.NavGraphDirections
+import org.mozilla.fenix.R
+import org.mozilla.fenix.browser.BrowserFragmentDirections
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+import org.mozilla.fenix.browser.browsingmode.BrowsingModeManager
+import org.mozilla.fenix.components.toolbar.ToolbarMenu
+import org.mozilla.fenix.components.toolbar.interactor.BrowserToolbarInteractor
+import org.mozilla.fenix.ext.nav
+import org.mozilla.fenix.tabstray.Page
+
+/**
+ * Navigation bar button types.
+ */
+sealed class NavBarButton {
+    /**
+     * A general action button, like Home or Back buttons.
+     */
+    data class ActionButton(val iconId: Int, val descriptionResourceId: Int) : NavBarButton() {
+        var onClick: () -> Unit = {}
+        var onLongPress: () -> Unit = {}
+    }
+
+    /**
+     * A tab counter.
+     */
+    object TabCounterButton : NavBarButton() {
+        var onClick: () -> Unit = {}
+    }
+
+    /**
+     * Three dot menu button.
+     */
+    data class ThreeDotMenuButton(val menuButton: MenuButton) : NavBarButton()
+}
+
+/**
+ * Holds a collection of default buttons.
+ *
+ * Items could be copied and customized for usage across the app.
+ * Also used for building previews.
+ */
+object DefaultButtons {
+    val home = NavBarButton.ActionButton(
+        iconId = R.drawable.mozac_ic_home_24,
+        descriptionResourceId = R.string.browser_toolbar_home,
+    )
+
+    val search = NavBarButton.ActionButton(
+        iconId = R.drawable.mozac_ic_search_24,
+        descriptionResourceId = R.string.browser_toolbar_home,
+    )
+
+    val menu = NavBarButton.ActionButton(
+        iconId = R.drawable.mozac_ic_ellipsis_vertical_24,
+        descriptionResourceId = R.string.mozac_browser_menu_button,
+    )
+
+    val back = NavBarButton.ActionButton(
+        iconId = R.drawable.mozac_ic_back_24,
+        descriptionResourceId = R.string.browser_menu_back,
+    )
+
+    val forward = NavBarButton.ActionButton(
+        iconId = R.drawable.mozac_ic_forward_24,
+        descriptionResourceId = R.string.browser_menu_forward,
+    )
+
+    val tabs = NavBarButton.TabCounterButton
+
+    val defaultItems = listOf(back, forward, home, tabs, menu)
+}
+
+/**
+ * Defines navigation bar locations with required dependencies for building the bav bar.
+ */
+sealed class NavBarLocation {
```

Comment:
now that `NavBarLocation` is only used in this file for `ActionItemListBuilder.build()`, do we even need to manage this complex data type anymore? Can this be replaced with direct calls to builder functions from HomeFragment & BrowserFragment?